From c392f4d1329b7def210ce96ceac0a5b4ee6c317e Mon Sep 17 00:00:00 2001
From: Eberhard Beilharz <eb1@sil.org>
Date: Tue, 10 Jan 2012 15:08:01 +0100
Subject: [PATCH] Xamarin-2462: Fix clipping of last line of dropdown

The height we calculate is the inner height, so we need to add the top
and bottom borders when setting the height of the listbox. The width
however is the outer width so we need to subtract the left and right
borders when setting the client width.

Change-Id: Ib60da79f820f4eca0f08271209a6eb9a46218c8f
---
 .../System.Windows.Forms/ComboBox.cs               |   14 ++++++++------
 1 files changed, 8 insertions(+), 6 deletions(-)

Index: fieldworks-mono/mcs/class/Managed.Windows.Forms/System.Windows.Forms/ComboBox.cs
===================================================================
--- fieldworks-mono.orig/mcs/class/Managed.Windows.Forms/System.Windows.Forms/ComboBox.cs
+++ fieldworks-mono/mcs/class/Managed.Windows.Forms/System.Windows.Forms/ComboBox.cs
@@ -2498,12 +2498,14 @@
 						vscrollbar_ctrl.Value = hli;
 					}
 				}
-				
-				Size = new Size (width, height);
-				textarea_drawable = ClientRectangle;
-				textarea_drawable.Width = width;
-				textarea_drawable.Height = height;
-				
+
+				var borderWidth = Hwnd.GetBorderWidth (CreateParams);
+				var borderAdjustment = dropdown_style == ComboBoxStyle.Simple ? new Size (0, 0) :
+					new Size (borderWidth.top + borderWidth.bottom, borderWidth.left + borderWidth.right);
+				Size = new Size (width, height + borderAdjustment.Height);
+				textarea_drawable = new Rectangle (ClientRectangle.Location,
+					new Size (width - borderAdjustment.Width, height));
+
 				if (vscrollbar_ctrl != null && show_scrollbar)
 					textarea_drawable.Width -= vscrollbar_ctrl.Width;
 
